src/main.rs

```diff
@@ -303,6 +306,25 @@ fn main() {
 
                         let final_frame_index = app.max_final_frame_index();
 
+                        //Check if prev_light_mode changed
+                        if prev_light_mode ^ app.ui_state.light_mode {
```

Comment:
Using `!=` instead of `^` more clearly expresses the intent here. If you're worried about the generated assembly, you can always double check in godbolt. The compiler is smart enough to just use xor here. https://rust.godbolt.org/z/brcoxYYbT